### PR TITLE
Replace PollImmediate with PollUntilContextTimeout

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Updates the polling functions in the patcher, replace, and scale packages to use wait.PollUntilContextTimeout instead of the deprecated wait.PollImmediate.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Can I remove this [line](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go#L389) also? not sure why it's there.
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go [hunk 1]
```diff
@@ -387,12 +388,13 @@ func (p *Patcher) deleteAndCreate(original runtime.Object, modified []byte, name
 		return modified, nil, err
 	}
 	// TODO: use wait
-	if err := wait.PollImmediate(1*time.Second, p.Timeout, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, p.Timeout, true, func(ctx context.Context) (bool, error) {
```
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go:391]: What is the reason of taking out from `if err` statement?.

File: test/utils/update_resources.go [hunk 2]
```diff
@@ -35,9 +36,9 @@ const (
 	waitRetryTimeout    = 5 * time.Minute
 )
 
-func RetryErrorCondition(condition wait.ConditionFunc) wait.ConditionFunc {
-	return func() (bool, error) {
-		done, err := condition()
+func RetryErrorCondition(condition wait.ConditionWithContextFunc) wait.ConditionWithContextFunc {
+	return func(context.Context) (bool, error) {
```
Comment (ardaguclu) [inline test/utils/update_resources.go:40]:
```suggestion
	return func(ctx context.Context) (bool, error) {
```

File: test/utils/update_resources.go [hunk 3]
```diff
@@ -35,9 +36,9 @@ const (
...
+		done, err := condition(context.Background())
```
Comment (ardaguclu) [inline test/utils/update_resources.go:41]:
```suggestion
		done, err := condition(ctx)
```

**Review by ardaguclu — COMMENTED** [review]
Thank you for taking care of this.

File: staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go (hunk 1 above)
Comment (omerap12) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go:391]: No particular reason. I can use the `if err` statement again if you'd prefer

Comment (ardaguclu) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go:391]: yes, please revert it back to `if err`

Comment (omerap12) [reply staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go]: reverted :)
btw what about this [line](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go#L389) should it be there?

**Comment by ardaguclu** [comment]
Thank you
/label tide/merge-method-squash
/lgtm
/approve

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**